src/NServiceBus.Core/Transports/ReceiveBehavior.cs

```diff
@@ -41,5 +42,37 @@ public Registration(): base("ReceiveMessage", typeof(ReceiveBehavior), "Try rece
         }
     }
 
-   
+    /// <summary>
+    /// The raw message coming from the transport
+    /// </summary>
+    public class ReceivedMessage
+    {
+        /// <summary>
+        /// The native id of the message
+        /// </summary>
+        public string MessageId { get; private set; }
+
+        /// <summary>
+        /// The message headers
+        /// </summary>
+        public Dictionary<string, string> Headers { get; private set; }
+        
+        /// <summary>
+        /// The message body
+        /// </summary>
+        public byte[] Body { get; private set; }
```

Comment:
One benefit would be that it opens up for things like http://www.philosophicalgeek.com/2015/02/06/announcing-microsoft-io-recycablememorystream/ to help lessen GC pressure and  LOH allocations. Not sure how much of an issue this is today but if we move into high throughput scenarios (High Speed sqltransport, zeromq etc) it might become more relevant?